src/v/cloud_storage/tests/delete_records_e2e_test.cc

```diff
@@ -613,3 +613,143 @@ FIXTURE_TEST(test_delete_from_archive_truncation, delete_records_e2e_fixture) {
       stm_manifest, stm_kafka_start);
     check_truncate_removes_override(size_above_override);
 }
+
+// Test that truncations applied to offsets that only exist in cloud storage
+// work as expected. Is similar to `test_delete_from_archive_truncation` with
+// the exception that everything is deleted from the local log before
+// truncations start.
+FIXTURE_TEST(test_no_local_delete_from_archive, delete_records_e2e_fixture) {
+    tests::remote_segment_generator gen(make_kafka_client().get(), *partition);
+    size_t records_per_seg = 5;
+    BOOST_REQUIRE_GE(
+      200,
+      gen.batches_per_segment(records_per_seg)
+        .num_segments(40)
+        .produce()
+        .get());
+    BOOST_REQUIRE(archiver->sync_for_tests().get());
+    archiver->apply_spillover().get();
+    auto& stm_manifest = archiver->manifest();
+    BOOST_REQUIRE_EQUAL(stm_manifest.get_spillover_map().size(), 3);
+    BOOST_REQUIRE_EQUAL(stm_manifest.size(), segs_per_spill);
+    kafka_delete_records_transport deleter(make_kafka_client().get());
+    deleter.start().get();
+
+    // The max collectible offset should of been set to the start of the local
+    // log above.
+    ss::abort_source as;
+    storage::housekeeping_config housekeeping_conf(
+      model::timestamp::min(),
+      0,
+      log->stm_manager()->max_collectible_offset(),
+      ss::default_priority_class(),
+      as);
+    partition->log()->housekeeping(housekeeping_conf).get();
+    // Wait for raft to write a snapshot and truncate.
+    tests::cooperative_spin_wait_with_timeout(10s, [this] {
+        return log->segments().size() == 1;
+    }).get();
+
+    auto assert_before_local_base = [this](model::offset kafka_offset) {
```

Comment:
nit: given how this is used, might make sense for it to be a kafka::offset, and cast in the comparison